src/spurt/graph/_delaunay.py

```diff
@@ -61,8 +67,115 @@ def boundary(self) -> np.ndarray:
     @property
     def links(self) -> np.ndarray:
         arcs: set[tuple[int, int]] = set()
-        for s in self.simplices:
-            arcs.add(order((s[0], s[1])))
-            arcs.add(order((s[0], s[2])))
-            arcs.add(order((s[1], s[2])))
+        for s in self.cycles:
+            arcs.add(order_points((s[0], s[1])))
+            arcs.add(order_points((s[0], s[2])))
+            arcs.add(order_points((s[1], s[2])))
         return np.array(sorted(arcs))
+
+
+class Reg2DGraph(PlanarGraphInterface):
+    """Class to hold 2D regular grid.
+
+    This is a utility class to test MCF implementation against other
+    regular 2D grid solvers. We will walk down the rows of the 2D grid
+    to determine the order of points, links and cycles.
+    """
+
+    def __init__(self, shape: tuple[int, int]):
+        """Create regular 2D graph of given shape."""
+        self._shape = shape
+
+    @property
+    def npoints(self) -> int:
+        return self._shape[0] * self._shape[1]
+
+    @property
+    def points(self) -> np.ndarray:
+        """Return points in 2D grid.
+
+        Points are returned in row order.
+        """
+        # We will just use indices here for position
+        pts = np.zeros((self.npoints, 2), dtype=int)
+        for ii in range(self._shape[0]):
+            i0 = ii * self._shape[1]
+            i1 = i0 + self._shape[1]
+
+            pts[i0:i1, 0] = np.arange(self._shape[1])
+            pts[i0:i1, 1] = ii
+
+        return pts
+
+    @property
+    def cycles(self) -> np.ndarray:
+        """Return rectangular loops in 2D grid."""
+        ncyc = self.npoints - self._shape[0] - self._shape[1] + 1
+        cyc = np.zeros((ncyc, 4), dtype=int)
+
+        ind = 0
+        # Iterate over the loops for top-left corner of loop
+        for ii in range(self._shape[0] - 1):
+            for jj in range(self._shape[1] - 1):
+                # cycles will go counter-clockwise like Delaunay
+                # top-left -> bottom-left -> bottom-right -> top-right
+                cyc[ind, :] = np.ravel_multi_index(
+                    ((ii, ii + 1, ii + 1, ii), (jj, jj, jj + 1, jj + 1)), self._shape
+                )
+                ind += 1
```

Comment:
There is no concept of grounding node on the original grid of `M x N` since the grounding node does not have any physical phase measurement associated with it. The grounding node comes into play in the dual graph of `(M-1) x (N-1)` cycles. It gets included in the dual graph in #6 . The grounding node in the dual graph plays the role of those cycles